test/testuninitvar.cpp

```diff
@@ -3683,6 +3683,17 @@ class TestUninitVar : public TestFixture {
                         "}", "test.c");
         ASSERT_EQUALS("", errout.str());
 
+        valueFlowUninit("int set(int *x);\n" // #12030
+                        "void foo(bool a) {\n"
+                        "    bool flag{0};\n"
+                        "    int x;\n"
+                        "    if (!a) {\n"
+                        "        flag = set(&x);\n"
+                        "    }\n"
+                        "    if (!flags && x==3) {}\n"
+                        "}\n");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
This should have unitialized warning because `!flags` is true so it go to `x == 3`.